docs/Contributing/API-for-contributors.md

```diff
@@ -668,19 +672,212 @@ Content-Type: application/octet-stream
+| ipados_team_id | integer | body | iPadOS hosts are automatically added to this team in Fleet when they appear in Apple Business Manager. If not specified, defaults to "No team" |
+
+#### Example
+
+`PATCH /api/v1/fleet/abm_tokens/1/teams`
+
+##### Request body
+
+```json
+{
+  "macos_team_id": 1,
+  "ios_team_id": 2,
+  "ipados_team_id": 3,
+}
+```
+
+##### Default response
+
+`Status: 200`
+
+```json
+{}
+```
+
+### Renew ABM token
+
+`PATCH /api/v1/fleet/abm_tokens/:id/renew`
+
+#### Parameters
+
+| Name | Type | In | Description |
+| ---- | ---- | -- | ----------- |
+| id | integer | path | *Required* The ABM token's ID |
+
+#### Example
+
+`PATCH /api/v1/fleet/abm_tokens/1/renew`
+
+##### Request header
+
+```http
+Content-Length: 850
+Content-Type: multipart/form-data; boundary=------------------------f02md47480und42y
+```
+
+##### Request body
+
+```http
+--------------------------f02md47480und42y
+Content-Disposition: form-data; name="token"; filename="server_token_abm.p7m"
+Content-Type: application/octet-stream
+
+<TOKEN_DATA>
+
+--------------------------f02md47480und42y
+```
+
+##### Default response
+
+`Status: 200`
+
+### Delete ABM token
+
+`DELETE /api/v1/fleet/abm_tokens/:id`
 
-### Disable automatic enrollment (ADE)
+#### Parameters
 
-`DELETE /api/v1/fleet/mdm/apple/abm_token`
+| Name | Type | In | Description |
+| ---- | ---- | -- | ----------- |
+| id | integer | path | *Required* The ABM token's ID |
 
 #### Example
 
-`DELETE /api/v1/fleet/mdm/apple/abm_token`
+`DELETE /api/v1/fleet/abm_tokens/1`
 
 ##### Default response
 
 `Status: 204`
 
+### Add VPP token
+
+`POST /api/v1/fleet/vpp_tokens`
+
+#### Parameters
+
+| Name | Type | In | Description |
+| ---- | ---- | -- | ----------- |
+| token | file | form | *Required* The file containing the content token (.vpptoken) from Apple Business Manager |
+
+#### Example
+
+`POST /api/v1/fleet/vpp_tokens`
+
+##### Request header
+
+```http
+Content-Length: 850
+Content-Type: multipart/form-data; boundary=------------------------f02md47480und42y
+```
+
+##### Request body
+
+```http
+--------------------------f02md47480und42y
+Content-Disposition: form-data; name="token"; filename="sToken_for_Acme.vpptoken"
+Content-Type: application/octet-stream
+<TOKEN_DATA>
+--------------------------f02md47480und42y
+```
+
+##### Default response
+
+`Status: 200`
+
+### Update VPP token's teams
+
+`PATCH /api/v1/fleet/vpp_tokens/:id/teams`
+
+#### Parameters
+
+| Name | Type | In | Description |
+| ---- | ---- | -- | ----------- |
+| id | integer | path | *Required* The ABM token's ID |
+| team_ids | list | body | If you choose specific teams, App Store apps in this VPP account will only be available to install on hosts in these teams. If not specified, defaults to all teams. |
+
+#### Example
+
+`PATCH /api/v1/fleet/vpp_tokens/1/teams`
+
+##### Request body
+
+```json
+{
+  "team_ids": [1, 2, 3]
```

Comment:
@noahtalerman to patch it back to 'all teams' I think it should be [] since 'no team' would be we just want to delete it then, thoughts?

Reply:
also arrays can be very brittle I would recommend changing this to 
```
{ "team_ids": {"macos_default": 1, "ios_default": 2, "ipados_default": 3}}
```
Gives us an opportunity to edit later w/o breaking array order and often it is more tech debt to make sure the order is correctly used in code down the line

Reply:
^ ignore, wrong location

Reply:
Nice catch. I forgot about all teams.

How do we solve this in existing API endpoints? I think let's be consistent with what we do in other API endpoints. If that's `[]` then makes sense.

Reply:
no team - 0

Reply:
null - not configured

Reply:
What about all teams?

Reply:
honestly thinking about it I think we need a second key `all_teams: true/false` alongside the team_ids. @noahtalerman thoughts?

Reply:
Hey! 

I think we want to add a reserved team for "All teams." There's a separate issue that tracks this here: #21264 

That's how other apps work.

> What about all teams?

Given that, how do other apps handle this? Do they add a reserved ID for the reserved team? Like we have for "No team" (`id=0`)

If that's the case, maybe we can avoid adding a new key.

I haven't built enough apps to know for sure. Tagging @lukeheath here too in case Luke has seen this before.

Reply:
Instead of adding another key, can we introduce the convention that passing an empty array for teams means "All teams"? Does that conflict with any of the other endpoints?

Reply:
In the current iteration we only support one team per VPP token, should we be passing in an array and validating that it only has one element or making a top level `team_id` integer?

Reply:
Moving forward with:
- `[0]` no team
- `null` not configured
- `[1]` team 1
- `[1, 2]` error, more than one team

Reply:
* [1,2] should work

Reply:
* [] all teams